src/tool.cpp

```diff
@@ -236,7 +248,23 @@ HitParams getHitParams(const ItemGroupList &armor_groups,
 				* armor / 100.0;
 	}
 
-	return {damage, 0};
+	for (const auto &groupcap : tp->groupcaps) {
+		const ToolGroupCap &cap = groupcap.second;
+
+		if (cap.uses != 0)
+			result_wear = 1.0 / cap.uses / pow(3.0, (double)(cap.maxlevel - 1.0));
```

Comment:
I don't understand why `cap.maxlevel - 1.0` is cast to a double instead of a float. Nevertheless, this line still needs a couple of code-style fixes:

```suggestion
			result_wear = 1.0f / cap.uses / pow(3.0f, (float)(cap.maxlevel - 1.0f));
```

Reply:
looking at the travis logs, it emits a warning that pow automatically upcasts to a double - should i change this back?